### PR TITLE
Adding AirVPN.pkg v2.8 Cask

### PR DESCRIPTION
**Pull request by moogar0880**
Adds support for installing the [AirVPN OS X](https://airvpn.org/macosx/) vpn client

### CODE REVIEW

File: Casks/airvpn.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  url "https://airvpn.org/repository/#{version}/airvpn_osx_x64_installer.pkg"
+  name 'Air VPN'
+  homepage 'https://airvpn.org/macosx/'
+  license :oss
+
+  binary '/Applications/AirVPN.app/Contents/MacOS/AirVPN'
+  pkg 'airvpn_osx_x64_installer.pkg'
+  uninstall :quit => 'com.airvpn.client',
+            :delete => '/Applications/AirVPN.app'
```
Comment (vitorgalvao) [inline Casks/airvpn.rb]: Are you sure moving this is the only thing the installer does? Have you got nothing with `pkgutil` scripts?

Comment (moogar0880) [inline Casks/airvpn.rb]: According to `pkgutil --files com.airvpn.client` that's the only thing it does. Don't quite know why but I wasn't able to get anything with `:pkgutil` working. It seemed like it only removed the `/usr/local/bin/AirVPN` symlink. 

If you have any suggestions I'd be more than willing to poke around with it some more, but so far this was the only way I was able to get it to actually remove AirVPN.app

Comment (vitorgalvao) [inline Casks/airvpn.rb]: What do you get with `$(brew --prefix)/Library/Taps/caskroom/homebrew-cask/developer/bin/list_recent_pkg_ids`?

Comment (moogar0880) [inline Casks/airvpn.rb]: I get the following
```
com.airvpn.client
com.adobe.pkg.FlashPlayer
com.intuit.turbotax.mac2014
org.virtualbox.pkg.virtualboxcli
org.virtualbox.pkg.vboxkexts
org.virtualbox.pkg.virtualbox
com.vagrant.vagrant
com.microsoft.office.all.core.pkg.14.4.8.update
com.microsoft.office.all.fix_permissions.pkg.14.4.8
com.microsoft.office.all.automator.pkg.14.4.8.update
```
It looks to me like `com.airvpn.client` is the only related package?

Comment (jawshooah) [inline Casks/airvpn.rb]: It's doing more than removing the symlink. Check it:

```
$ brew cask uninstall airvpn --verbose --debug
==> Creating directories
==> Uninstalling Cask airvpn
==> Loading Cask definitions
==> Translating 'airvpn' into a valid Cask source
==> Testing source class Hbc::Source::URI
==> Testing source class Hbc::Source::PathSlashRequired
==> Testing source class Hbc::Source::TappedQualified
==> Testing source class Hbc::Source::UntappedQualified
==> Testing source class Hbc::Source::Tapped
==> Success! Using source class Hbc::Source::Tapped
==> Resolved Cask URI or file source to '/usr/local/Library/Taps/caskroom/homebrew-cask/Casks/airvpn.rb'
==> Cask instance dumps in YAML:
==> Cask instance toplevel:
--- !ruby/object:KlassPrefixAirvpn
sourcefile_path: !ruby/object:Pathname
  path: /usr/local/Library/Taps/caskroom/homebrew-cask/Casks/airvpn.rb
token: airvpn
==> Cask instance method 'name':
---
- Air VPN
==> Cask instance method 'homepage':
--- https://airvpn.org/macosx/
...
==> Cask instance method 'url':
--- !ruby/object:Hbc::URL
uri: !ruby/object:URI::HTTPS
  scheme: https
  user:
  password:
  host: airvpn.org
  port: 443
  path: /repository/2.8/airvpn_osx_x64_installer.pkg
  query:
  opaque:
  registry:
  fragment:
  parser:
user_agent:
cookies:
referer:
using:
revision:
trust_cert:
data:
==> Cask instance method 'appcast':
---
...
==> Cask instance method 'version':
--- '2.8'
==> Cask instance method 'license':
--- !ruby/object:Hbc::DSL::License
value: :oss
==> Cask instance method 'tags':
---
...
==> Cask instance method 'sha256':
--- 311babbdecef814a8c6168ee625990338489dde973bef83b3fe157db493a1cef
...
==> Cask instance method 'artifacts':
---
:binary: !ruby/object:Set
  hash:
    ? - /Applications/AirVPN.app/Contents/MacOS/AirVPN
    : true
:pkg: !ruby/object:Set
  hash:
    ? - airvpn_osx_x64_installer.pkg
    : true
:uninstall: !ruby/object:Set
  hash:
    ? :pkgutil: com.airvpn.client
    : true
==> Cask instance method 'caveats':
--- []
==> Cask instance method 'depends_on':
---
...
==> Cask instance method 'conflicts_with':
---
...
==> Cask instance method 'container':
---
...
==> Cask instance method 'gpg':
---
...
==> Cask instance method 'accessibility_access':
---
...
==> Hbc::Installer.uninstall
==> Un-installing artifacts
==> Determining which artifacts are present in Cask airvpn
==> Checking for artifact class Hbc::Artifact::PreflightBlock
==> Checking for artifact class Hbc::Artifact::NestedContainer
==> Checking for artifact class Hbc::Artifact::Installer
==> Checking for artifact class Hbc::Artifact::App
==> Checking for artifact class Hbc::Artifact::Suite
==> Checking for artifact class Hbc::Artifact::Artifact
==> Checking for artifact class Hbc::Artifact::Colorpicker
==> Checking for artifact class Hbc::Artifact::Pkg
==> Checking for artifact class Hbc::Artifact::Prefpane
==> Checking for artifact class Hbc::Artifact::Qlplugin
==> Checking for artifact class Hbc::Artifact::Font
==> Checking for artifact class Hbc::Artifact::Service
==> Checking for artifact class Hbc::Artifact::StageOnly
==> Checking for artifact class Hbc::Artifact::Binary
==> Checking for artifact class Hbc::Artifact::InputMethod
==> Checking for artifact class Hbc::Artifact::InternetPlugin
==> Checking for artifact class Hbc::Artifact::ScreenSaver
==> Checking for artifact class Hbc::Artifact::Uninstall
==> Checking for artifact class Hbc::Artifact::PostflightBlock
==> Checking for artifact class Hbc::Artifact::Zap
==> 3 artifact/s defined
Hbc::Artifact::Pkg
Hbc::Artifact::Binary
Hbc::Artifact::Uninstall
==> Un-installing artifact of class Hbc::Artifact::Pkg
==> Un-installing artifact of class Hbc::Artifact::Binary
==> Removing Binary symlink: '/usr/local/bin/AirVPN'
==> Un-installing artifact of class Hbc::Artifact::Uninstall
==> Running uninstall process for airvpn; your password may be necessary
==> Removing files from pkgutil Bill-of-Materials
==> Executing: ["/usr/sbin/pkgutil", "--pkgs=com.airvpn.client"]
==> Deleting pkg files
==> Executing: ["/usr/sbin/pkgutil", "--only-files", "--files", "com.airvpn.client"]
==> Executing: ["/usr/sbin/pkgutil", "--pkg-info-plist", "com.airvpn.client"]
==> Executing: ["/usr/sbin/pkgutil", "--pkg-info-plist", "com.airvpn.client"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rm", "-f", "--", "#<Pathname:/Applications/AirVPN.app/Contents/Info.plist>", "#<Pathname:/Applications/AirVPN.app/Contents/MacOS/AirVPN>", "#<Pathname:/Applications/AirVPN.app/Contents/MacOS/openvpn>", "#<Pathname:/Applications/AirVPN.app/Contents/MacOS/stunnel>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/AirVPN.exe>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/AirVPN.exe.mdb>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/Lib.Core.dll>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/Lib.Core.dll.mdb>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/Mono.Security.dll>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/Platforms.Osx.dll>", "#<Pathname:/Applications/AirVPN.app/Conten...
==> Deleting pkg symlinks and special files
==> Executing: ["/usr/sbin/pkgutil", "--files", "com.airvpn.client"]
==> Executing: ["/usr/sbin/pkgutil", "--only-files", "--files", "com.airvpn.client"]
==> Executing: ["/usr/sbin/pkgutil", "--only-dirs", "--files", "com.airvpn.client"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rm", "-f", "--", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/libexpat.1.dylib>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/libgdiplus.dylib>", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle/libgif.4.dylib>"]
==> Deleting pkg directories
==> Executing: ["/usr/sbin/pkgutil", "--only-dirs", "--files", "com.airvpn.client"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/chmod", "--", "777", "#<Pathname:/Applications/AirVPN.app/Contents/_CodeSignature>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rmdir", "--", "#<Pathname:/Applications/AirVPN.app/Contents/_CodeSignature>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/chmod", "--", "777", "#<Pathname:/Applications/AirVPN.app/Contents/Resources>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rmdir", "--", "#<Pathname:/Applications/AirVPN.app/Contents/Resources>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/chmod", "--", "777", "#<Pathname:/Applications/AirVPN.app/Contents/MacOS>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rmdir", "--", "#<Pathname:/Applications/AirVPN.app/Contents/MacOS>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/chmod", "--", "777", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rmdir", "--", "#<Pathname:/Applications/AirVPN.app/Contents/MonoBundle>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/chmod", "--", "777", "#<Pathname:/Applications/AirVPN.app/Contents>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rmdir", "--", "#<Pathname:/Applications/AirVPN.app/Contents>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/chmod", "--", "777", "#<Pathname:/Applications/AirVPN.app>"]
==> Executing: ["/usr/bin/sudo", "-E", "--", "/bin/rmdir", "--", "#<Pathname:/Applications/AirVPN.app>"]
==> Unregistering pkg receipt (aka forgetting)
==> Executing: ["/usr/bin/sudo", "-E", "--", "/usr/sbin/pkgutil", "--forget", "com.airvpn.client"]
==> Purging files for version 2.8 of Cask airvpn
```

Comment (vitorgalvao) [inline Casks/airvpn.rb]: > It looks to me like `com.airvpn.client` is the only related package?

Exactly. That means this line should be replaced with `:pkgutil => 'com.airvpn.client'`.

File: Casks/airvpn.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  sha256 '311babbdecef814a8c6168ee625990338489dde973bef83b3fe157db493a1cef'
+
+  url "https://airvpn.org/repository/#{version}/airvpn_osx_x64_installer.pkg"
+  name 'Air VPN'
+  homepage 'https://airvpn.org/macosx/'
+  license :oss
+
+  binary '/Applications/AirVPN.app/Contents/MacOS/AirVPN'
+  pkg 'airvpn_osx_x64_installer.pkg'
+  uninstall :quit => 'com.airvpn.client',
```
Comment (vitorgalvao) [inline Casks/airvpn.rb]: There should be an empty line above `uninstall`.

File: Casks/airvpn.rb (hunk 1 above)
Comment (moogar0880) [inline Casks/airvpn.rb]: Huh, I must have goofed when I was playing around with `:pkgutil`, previously it didn't appear to be cleaning up `AirVPN.app`, but you're right, it looks like that does the trick. I'll replace the `:delete` with a `:pkgutil`.

Thanks for all the help guys

Comment (vitorgalvao) [inline Casks/airvpn.rb]: Thank you for all your work on this. Could you please [squash your commits](http://davidwalsh.name/squash-commits-git)? @mention me after you do it, so I get a warning to check back.

**Comment by moogar0880** [comment]
@vitorgalvao squashed my commits, looks like things should be good to merge. Thanks again for your help on this

**Comment by vitorgalvao** [comment]
Looks great. Thank you.

**Pull request merged**